lib/ddtrace/contrib/aws/instrumentation.rb

```diff
@@ -0,0 +1,43 @@
+module Datadog
+  module Contrib
+    module Aws
+      # A Seahorse::Client::Plugin that enables instrumentation for all AWS services
+      class Instrumentation < Seahorse::Client::Plugin
+        def add_handlers(handlers, _)
+          handlers.add(Handler, step: :validate)
+        end
+      end
+
+      # Generates Spans for all interactions with AWS
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          pin = Datadog::Pin.get_from(::Aws)
+
+          return @handler.call(context) unless pin && pin.tracer
```

Comment:
we should check `pin.enabled?`: https://github.com/DataDog/dd-trace-rb/blob/54ed5b7212e506f9b8304ee0361d2052f3eaf01e/lib/ddtrace/pin.rb#L32-L35